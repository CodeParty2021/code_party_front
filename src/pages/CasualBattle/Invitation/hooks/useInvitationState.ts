import { useEffect, useState } from "react";
import { Params, useNavigate, useParams } from "react-router-dom";
import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import { RootState } from "store";
import { useSelector } from "react-redux";

export type IResponse = {
  isLogin: boolean;
  errorMessage?: string;
  roomId?: string;
  enterBtnDisabled: boolean;
  enterBtnClickHandler: () => void;
};

export const useInvitationState = (): IResponse => {
  const { roomId } = useParams<Params<string>>();
  const { isLogin } = useSelector((state: RootState) => state.user);
  const [enterBtnDisabled, setEnterBtnDisabled] = useState(false);
  const { room, enterRoom } = useRoomSync();
  const [errorMessage, setErrorMessage] = useState<string | undefined>(
    undefined
  );
  const navigate = useNavigate();

  useEffect(() => {
    if (room.isEntered) {
      navigate("/casual-battle/waiting-room");
    }
  }, [room.isEntered]);

  //入場ボタン
  const _enterBtnHandler = () => {
    if (typeof roomId === "string") {
      setEnterBtnDisabled(true);
      enterRoom(roomId).catch(() => {
        setEnterBtnDisabled(false);
      });
    } else {
      setErrorMessage("招待リンクが無効です。");
    }
  };

  return {
    isLogin: isLogin,
    errorMessage: errorMessage,
    roomId: roomId,
    enterBtnDisabled: enterBtnDisabled,
    enterBtnClickHandler: _enterBtnHandler,
  };
};
