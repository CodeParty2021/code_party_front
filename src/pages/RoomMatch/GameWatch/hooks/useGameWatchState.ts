import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import { useFetchJson } from "hooks/ResultAPIHooks/useFetchJson";
import { useUnityGame } from "hooks/UnityGameHooks/useUnityGame";
import { UnityContext } from "react-unity-webgl";
import { useDummyLoading } from "hooks/DummyLoadingHooks/useDummyLoading";

export type IResponse = {
  unityContext: UnityContext;
  state: "Analyzing" | "Running" | "Finish" | "AnalyzingError";
  messageType: "Analyzing" | "AnalyzingError";
  exitBtnHandler: () => void;
};

export const useGameWatchState = (): IResponse => {
  const { room, updateMember } = useRoomSync();
  const { data: json, fetchJson } = useFetchJson();
  const { unityContext, startGame, unityStatus } = useUnityGame("SquarePaint");
  const [messageType, setMessageType] =
    useState<IResponse["messageType"]>("Analyzing");
  const { dummyLoadingState, startDummyLoad } = useDummyLoading(1000);
  const navigate = useNavigate();

  const state = room.info?.analyzingResult?.error
    ? "AnalyzingError"
    : !room.info?.analyzingResult?.resultId ||
      unityStatus.isLoading ||
      dummyLoadingState.isLoading
    ? "Analyzing"
    : unityStatus.isRunning
    ? "Running"
    : "Finish";

  // メッセージの操作
  useEffect(() => {
    if (state === "Analyzing") setMessageType("Analyzing");
    else if (state === "AnalyzingError") setMessageType("AnalyzingError");
  }, [state]);

  //初期処理
  useEffect(() => {
    //ユーザ状態を更新
    updateMember({
      status: "watching",
    });

    // ダミーロード開始
    startDummyLoad();
  }, []);

  // 何らかの理由でルームから出された場合はロビーに戻る
  useEffect(() => {
    if (!room.isEntered) {
      navigate("/room-match");
    }
  }, [room.isEntered]);

  //シミュレーション結果が更新されたら結果をフェッチする
  useEffect(() => {
    if (room.info?.analyzingResult?.resultId) {
      fetchJson(room.info?.analyzingResult.resultId);
    }
  }, [room.info?.analyzingResult]);

  // 観戦開始
  useEffect(() => {
    if (!unityStatus.isLoading && json) {
      startGame(json);
    }
  }, [unityStatus.isLoading, json]);

  const _exitBtnHandler = () => {
    navigate("/room-match/waiting-room");
  };

  return {
    state,
    messageType,
    unityContext,
    exitBtnHandler: _exitBtnHandler,
  };
};
