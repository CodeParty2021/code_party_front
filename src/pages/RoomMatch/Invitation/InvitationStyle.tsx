import styled from "styled-components";
import { GRAY_80, GRAY_90, WHITE } from "styles/colors";

export const SignInButton = styled.button`
  width: 306px;
  background-color: ${WHITE};
  display: flex;
  margin: 4px;
  height: 56px;
  border-radius: 8px;
  text-align: left;
  text-align: center;
  padding: 12px 32px;
`;

export const IconArea = styled.div`
  display: block;
  margin: auto 24px auto 0;
  height: 24px;
  text-align: center;
`;

export const ButtonText = styled.p`
  margin: auto auto auto 0;
  font-weight: 700;
  color: ${GRAY_80};
`;

export const Modal = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  z-index: 51;
  background: ${WHITE};

  box-shadow: 0px 4px 16px -2px rgba(40, 45, 62, 0.1);
  border-radius: 32px;
  padding: 32px 40px 32px;

  display: flex;
  flex-direction: column;
  gap: 24px;
  min-width: 1200px;
`;

export const Description = styled.div`
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 26px;
  /* identical to box height */

  font-feature-settings: "palt" on;
  color: ${GRAY_90};
`;
export const Title = styled.div`
  font-family: "Noto Sans JP";
  font-style: normal;
  font-weight: 700;
  font-size: 24px;
  line-height: 35px;
  /* identical to box height */

  font-feature-settings: "palt" on;
  color: ${GRAY_90};

  transform: matrix(1, 0, -0.08, 1, 0, 0);
`;
