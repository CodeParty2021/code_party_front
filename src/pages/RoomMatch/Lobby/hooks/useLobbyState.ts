import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import { RefObject, useCallback, useEffect, useMemo, useRef, useState } from "react";
import { useNavigate } from "react-router-dom";

export type IResponse = {
  isLoading: boolean;
  errorMessage: string;
  roomIdRef: RefObject<HTMLInputElement>;
  createRoomHandler: () => void;
  createRoomDisabled: boolean;
  enterRoomHandler: () => Promise<void>;
  enterRoomDisabled: boolean;
  backButtonHandler: () => void;
};

export const useLobbyState = (): IResponse => {
  const [isLoading, setIsLoading] = useState<boolean>(false);
  const [errorMessage, setErrorMessage] = useState<string>("");
  const [isProcessingAny, setIsProcessingAny] = useState<boolean>(false);
  const [succeedEnterRoom, setSucceedEnterRoom] = useState<boolean>(false);
  const { room, createRoom, enterRoom } = useRoomSync();
  const navigate = useNavigate();
  const roomIdRef = useRef<HTMLInputElement>(null!);

  const createEnterButtonDisabled = useMemo(() => (
    isProcessingAny || room.isEntered || succeedEnterRoom
  ), [isProcessingAny, room.isEntered]);

  useEffect(() => {
    if (room.isEntered) {
      navigate("/room-match/waiting-room");
    }
  }, [room.isEntered]);

  // ルーム
  const _createRoomHandler = useCallback(async () => {
    setIsLoading(true);
    setIsProcessingAny(true);
    await createRoom()
      .then(() => {
        setSucceedEnterRoom(true);
      })
      .catch(() => {
        navigate("/error");
      });
    setIsProcessingAny(false);
  }, []);

  const _enterRoomHandler = useCallback(async () => {
    const value = roomIdRef.current?.value;

    setIsProcessingAny(true);
    if (typeof value === "string") {
      if (value == "") {
        setErrorMessage("値を入力してください。");
      } else {
        await new Promise((resolve) => setTimeout(resolve, 500));
        await enterRoom(value).then(() => {
          setSucceedEnterRoom(true);
        }).catch((e) => {
          if (e.message == "roomId is empty") {
            setErrorMessage("値を入力してください。");
          } else if (e.message == "roomId is not found") {
            setErrorMessage("ルームIDが無効です。");
          } else if (e.message == "room is full") {
            setErrorMessage("ルームが満員です。");
          }
        });
      }
    } else {
      setErrorMessage("入力が不正です。");
    }
    setIsProcessingAny(false);
  }, []);

  // 戻るボタン
  const _backButtonHandler = useCallback(() => {
    navigate("/select-mode");
  }, []);

  return {
    isLoading,
    errorMessage,
    roomIdRef,
    createRoomHandler: _createRoomHandler,
    createRoomDisabled: createEnterButtonDisabled,
    enterRoomHandler: _enterRoomHandler,
    enterRoomDisabled: createEnterButtonDisabled,
    backButtonHandler: _backButtonHandler,
  };
};
