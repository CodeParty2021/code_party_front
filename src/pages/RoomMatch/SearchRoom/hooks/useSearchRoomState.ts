import { useEffect, useState } from "react";
import { useNavigate } from "react-router-dom";

import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";

export type IResponse = {
  roomIdTextBoxValue: string;
  roomIdTextBoxChangeHandler: (value: string) => void;
  enterBtnDisabled: boolean;
  enterBtnClickHandler: () => void;
};

export const useSearchRoomState = (): IResponse => {
  const { room, enterRoom } = useRoomSync();
  const [roomId, setRoomId] = useState("");
  const [enterBtnDisabled, setEnterBtnDisabled] = useState(false);
  const navigate = useNavigate();

  useEffect(() => {
    if (room.isEntered) {
      navigate("/room-match/waiting-room");
    }
  }, [room.isEntered]);

  const _roomIdTextBoxChangeHandler = (value: string) => {
    setRoomId(value);
  };

  const _enterBtnHandler = () => {
    setEnterBtnDisabled(true);
    enterRoom(roomId).catch(() => {
      setEnterBtnDisabled(false);
    });
  };

  return {
    roomIdTextBoxValue: roomId,
    roomIdTextBoxChangeHandler: _roomIdTextBoxChangeHandler,
    enterBtnDisabled: enterBtnDisabled,
    enterBtnClickHandler: _enterBtnHandler,
  };
};
