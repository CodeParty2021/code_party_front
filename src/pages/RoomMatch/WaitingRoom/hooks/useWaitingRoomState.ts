import { useEffect, useState, useMemo } from "react";
import { useNavigate, useSearchParams } from "react-router-dom";

import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import { CodeType, useFetchCodes } from "hooks/CodeAPIHooks/useFetchCodes";
import { UserState, UserAction } from "services/RoomSync/RoomSync";
import { RootState } from "store";
import { useSelector } from "react-redux";
import { User } from "services/user/user";

export type IResponse = {
  user: User | null;
  roomInfo: {
    roomId: string;
    invitationLink: string;
    host: UserState;
    memberKeys: string[];
    members: { [id: string]: UserState };
    actionKeys: string[];
    actions: { [id: string]: UserAction };
  };
  isHost: boolean;
  status?: "waiting" | "watching";
  ready: boolean;

  readyBtnHandler: () => void;
  readyBtnDisabled: boolean;
  exitBtnHandler: () => void;
  startBtnDisabled: boolean;
  startBtnHandler: () => void;
  isCopyBtnClicked: boolean;
  invitationBtnHandler: () => void;
  kickUserHandler: (userId: string) => void;
  code: {
    codes: CodeType[];
    loading: boolean;
  };
  selectedCode: CodeType | null;
  onChangeSelectedCode: (code: CodeType) => void;
  onStartToEditCode: () => void;
  showCodeSelectModal: boolean;
  openCodeSelectModal: () => void;
  closeCodeSelectModal: () => void;
};

export const useWaitingRoomState = (): IResponse => {
  const {
    room,
    isHost,
    updateRoomInfo,
    updateMember,
    updateOtherMember,
    exitRoom,
  } = useRoomSync();
  const { data: codes = [], loading } = useFetchCodes();
  const [preRoomStatus, setPreRoomStatus] = useState<"waiting" | "watching">(
    "watching"
  );
  const [selectedCode, setSelectedCode] = useState<CodeType | null>(null);
  const [ready, setReady] = useState(false);
  const [readyBtnDisabled, setReadyBtnDisabled] = useState(true);
  const [startBtnDisabled, setStartBtnDisabled] = useState(true);
  const [isCopyBtnClicked, setIsCopyBtnClicked] = useState(false);
  let { user } = useSelector((state: RootState) => state.user);
  const navigate = useNavigate();
  const [searchParams] = useSearchParams();
  const dummyUser: UserState = {
    displayName: "",
    status: "disconnect",
    ready: false,
  };

  const modalOn = useMemo(
    () => (searchParams.get("modal") === "on" ? true : false),
    []
  );

  // ユーザ状態を更新
  useEffect(() => {
    updateMember({
      status: "waiting",
    });
  }, []);

  // DB上で退出処理が完了したら退出
  useEffect(() => {
    if (!room.isEntered) {
      navigate("/room-match");
    }
  }, [room.isEntered]);

  // 準備ボタンの有効化
  useEffect(() => {
    //コードが選択されたら準備完了ボタンを押せるようにする
    if (selectedCode != null) {
      setReadyBtnDisabled(false);
    }
  }, [selectedCode]);

  // ルームステータスがwaitingからwatchingになったら観戦画面に遷移
  useEffect(() => {
    if (room.info) {
      if (preRoomStatus == "waiting" && room.info.status == "watching") {
        navigate("/room-match/result");
      }
      setPreRoomStatus(room.info?.status);
    }
  }, [room.info?.status]);

  // ready状態をDBに反映
  useEffect(() => {
    updateMember({
      ready: ready,
      codeId: selectedCode ? selectedCode.id : "",
    });
  }, [ready]);

  // クエリパラメータにmodal=onを渡されたらモーダルを表示
  useEffect(() => {
    if (modalOn) {
      setShowCodeSelectModal(true);
    }
  }, [modalOn]);

  // ホストの処理
  useEffect(() => {
    if (isHost) {
      if (room.info && room.info.status == "waiting") {
        // 全員が準備完了になったら開始ボタンを有効にする
        let newDisabled = false;
        for (let key of Object.keys(room.members)) {
          if (!room.members[key].ready) {
            newDisabled = true;
            break;
          }
        }
        setStartBtnDisabled(newDisabled);
      }
      // 全員がWaitingRoomに返ってきたらroomの状態をwaitingに更新する
      let newRoomStatus: "waiting" | "watching" = "waiting";
      for (let key of Object.keys(room.members)) {
        if (room.members[key].status == "watching") {
          newRoomStatus = "watching";
        }
      }
      if (newRoomStatus == "waiting" && room.info?.status == "watching") {
        updateRoomInfo({
          status: "waiting",
          analyzingResult: null,
        });
      }
    }
  }, [room.members, isHost]);

  // kick処理 member.status = kickingの場合そのユーザをキックする
  useEffect(() => {
    if (user) {
      for (let key of Object.keys(room.members)) {
        if (room.members[key].status == "kicking" && user.id == key) {
          exitRoom();
        }
      }
    }
  }, [room.members]);

  // ---- イベントハンドラ ---- //

  const _readyBtnHandler = () => {
    setReady(!ready);
  };

  const _exitBtnHandler = () => {
    exitRoom();
  };

  const _startBtnHandler = () => {
    updateRoomInfo({
      status: "watching",
    });
  };

  const _onChangeSelectedCode = (code: CodeType) => {
    setSelectedCode(code);
  };

  const _onStartToEditCode = () => {
    if (selectedCode) {
      navigate(`/free-coding/${selectedCode.id}/selectCode`);
    }
  };

  //invitationURLのコピーボタン
  const _invitationBtnHandler = () => {
    if (room.invitationLink) {
      navigator.clipboard.writeText(room.invitationLink).then(() => {
        setIsCopyBtnClicked(true);
      });
    }
  };
  const _kickUserBtnHandler = (userId: string) => {
    updateOtherMember(userId, { status: "kicking" });
  };

  //モーダル表示非表示
  const [showCodeSelectModal, setShowCodeSelectModal] =
    useState<boolean>(false);
  const openCodeSelectModal = () => {
    setShowCodeSelectModal(true);
  };
  const closeCodeSelectModal = () => {
    setShowCodeSelectModal(false);
  };

  return {
    user: user,
    roomInfo: {
      roomId: room.id ? room.id : "",
      invitationLink: room.invitationLink ? room.invitationLink : "",
      host:
        room.info && room.info.host in room.members
          ? room.members[room.info.host]
          : dummyUser,
      memberKeys: room.sortedKeysOfMembers,
      members: room.members,
      actionKeys: room.sortedKeysOfActions,
      actions: room.actions,
    },
    isHost: isHost,
    status: room.info?.status,
    ready: ready,
    readyBtnHandler: _readyBtnHandler,
    readyBtnDisabled: readyBtnDisabled,
    exitBtnHandler: _exitBtnHandler,
    startBtnDisabled: startBtnDisabled,
    startBtnHandler: _startBtnHandler,
    isCopyBtnClicked: isCopyBtnClicked,
    invitationBtnHandler: _invitationBtnHandler,
    kickUserHandler: _kickUserBtnHandler,
    code: {
      codes: codes,
      loading: loading,
    },
    selectedCode: selectedCode,
    onChangeSelectedCode: _onChangeSelectedCode,
    onStartToEditCode: _onStartToEditCode,
    showCodeSelectModal,
    openCodeSelectModal,
    closeCodeSelectModal,
  };
};
