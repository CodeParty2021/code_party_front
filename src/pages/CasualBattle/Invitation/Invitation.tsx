import React from "react";
import { SignInScreen } from "components/SignInScreen/SignInScreen";
import { useInvitationState } from "./hooks/useInvitationState";
import { Head } from "components/Head/Head";

type Props = {};

export const CasualBattleInvitation: React.FC<Props> = () => {
  const {
    isLogin,
    errorMessage,
    roomId,
    enterBtnDisabled,
    enterBtnClickHandler,
    hostId,
  } = useInvitationState();

  if (typeof roomId === "string") {
    // if not login yet, Go to SignIn
    if (isLogin == false) {
      return (
        <>
          <Head
            title="招待されています"
            ogDescription={`HostId:${hostId}からRoomId:${roomId}に招待されています。`}
            ogType="article"
          />
          <SignInScreen
            signInSuccessUrl={`/casual-battle/invitation/${roomId}`}
          />
        </>
      );
    }
    //display enter button
    else {
      return (
        <div>
          <Head
            title="招待されています"
            ogDescription={`HostId:${hostId}からRoomId:${roomId}に招待されています。`}
            ogType="article"
          />
          <h1>次のルームに招待されています。</h1>
          <p>RoomID:{roomId}</p>
          <button
            onClick={enterBtnClickHandler}
            disabled={enterBtnDisabled}
            id="enter-btn"
          >
            入場する
          </button>
        </div>
      );
    }
  } else {
    return <p>{errorMessage || "招待リンクが無効です"} </p>;
  }
};
