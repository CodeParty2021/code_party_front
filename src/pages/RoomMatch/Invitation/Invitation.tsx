import React from "react";
import { SignInScreen } from "components/SignInScreen/SignInScreen";
import { useInvitationState } from "./hooks/useInvitationState";
import { Head } from "components/Head/Head";

type Props = {};

export const RoomMatchInvitation: React.FC<Props> = () => {
  const {
    isLogin,
    errorMessage,
    roomId,
    enterBtnDisabled,
    enterBtnClickHandler,
    hostName,
  } = useInvitationState();
  const header = (
    <Head
      title="ALGOSMO"
      ogDescription={
        hostName
          ? `${hostName}からカジュアル対戦に招待されています！`
          : "ルーム情報が見つかりませんでした。"
      }
      ogType="article"
      ogImage="https://codeparty.netlify.app/img/modeselectcard_battle.png"
      ogImageHeight="412"
      ogImageWidth="616"
    />
  );

  if (typeof roomId === "string") {
    // if not login yet, Go to SignIn
    if (isLogin == false) {
      return (
        <>
          {header}
          <SignInScreen signInSuccessUrl={`/room-match/invitation/${roomId}`} />
        </>
      );
    }
    //display enter button
    else {
      return (
        <div>
          {header}
          <h1>次のルームに招待されています。</h1>
          <p>RoomID:{roomId}</p>
          <button
            onClick={enterBtnClickHandler}
            disabled={enterBtnDisabled}
            id="enter-btn"
          >
            入場する
          </button>
        </div>
      );
    }
  } else {
    return <p>{errorMessage || "招待リンクが無効です"} </p>;
  }
};
