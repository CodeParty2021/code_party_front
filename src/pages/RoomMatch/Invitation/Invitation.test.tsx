import React from "react";
import { shallow } from "enzyme";

import { RoomMatchInvitation } from "./Invitation";
import { IResponse, useInvitationState } from "./hooks/useInvitationState";
import { useFirebaseAuth } from "hooks/FirebaseAuthHooks/useFirebaseAuthHooks";

jest.mock("./hooks/useInvitationState");
jest.mock("hooks/FirebaseAuthHooks/useFirebaseAuthHooks");

const useSelectorMock = useInvitationState as jest.Mock;

const useFirebaseAuthMock = useFirebaseAuth as jest.Mock;

const state: IResponse = {
  isLogin: true,
  errorMessage: undefined,
  roomId: "sampleID",
  enterBtnDisabled: true,
  enterBtnClickHandler: jest.fn(),
  hostName: "hostName",
};

describe("<RoomMatchInvitation />", () => {
  beforeEach(() => {
    useSelectorMock.mockReturnValue({
      ...state,
    });
    useFirebaseAuthMock.mockReturnValue({
      signInOfFirebase: jest.fn(),
    });
  });
  afterEach(() => {
    jest.resetAllMocks();
  });

  it("snapshot test", () => {
    const wrapper = shallow(<RoomMatchInvitation />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
