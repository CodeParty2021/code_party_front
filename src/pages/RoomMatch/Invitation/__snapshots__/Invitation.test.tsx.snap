// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomMatchInvitation /> snapshot test 1`] = `
Array [
  <div>
    <StarBackground />
    <Head
      ogDescription="hostNameからルーム対戦に招待されています！"
      ogImage="https://codeparty.netlify.app/img/modeselectcard_battle.png"
      ogImageHeight="412"
      ogImageWidth="616"
      ogType="article"
      title="ALGOSMO"
    />
    <styled.div>
      <styled.div>
        hostNameからルーム対戦に招待されています！
      </styled.div>
      <styled.div>
        RoomID:
        sampleID
      </styled.div>
      <Button
        color="blue"
        icon={null}
        onClick={[MockFunction]}
        size="L"
        status="disabled"
        value="入室する"
      />
    </styled.div>
  </div>,
]
`;
